### ISSUE TITLE
rust-analyzer.linkedProjects Warning when opening VSCode projects

### ISSUE THREAD

**Issue by lookbusy1344** [rust-lang/rust-analyzer#17013]
**rust-analyzer version**: 0.3.1906

**rustc version**: stable-x86_64-pc-windows-msvc unchanged - rustc 1.77.1 (7cf61ebde 2024-03-27)

**editor or extension**: vscode

**relevant settings**: just defaults

With rust-analyser (0.3.1906 released this week) and vscode I’m getting an error about my `cargo.toml` file in several projects. These are projects that have worked fine for many months. The recommended fix is to add `settings.json` containing:

```
{
    "rust-analyzer.linkedProjects": [
        ".\\Cargo.toml"
    ]
}
```

Every time I re-open the project it reports the same error, and so the settings file becomes:

```
{
    "rust-analyzer.linkedProjects": [
        ".\\Cargo.toml",
        ".\\Cargo.toml"
    ]
}
```

The last version free of the bug seems to be 0.3.1885

Platform is “stable-x86_64-pc-windows-msvc unchanged - rustc 1.77.1 (7cf61ebde 2024-03-27)”

**Comment by lnicola** [comment]
What do your projects look like?

I think I got that "file is outside of the project" warning a couple of times lately, but it doesn't say which file it was, but I switched to another tab and I couldn't reproduce it afterwards.

**Comment by lookbusy1344** [comment]
As fair as I’m aware it’s just a vanilla cargo layout. The repo is at

https://github.com/lookbusy1344/HashRust

I see the message (image attached) every time I open the project. I have the 4 .rs files open, with hasher.rs focused.

![IMG_1417](https://github.com/rust-lang/rust-analyzer/assets/3680611/af3301e4-7a96-43ef-9b2d-94600822ec4f)
![IMG_1416](https://github.com/rust-lang/rust-analyzer/assets/3680611/43b53d77-bc81-4de5-b53a-e66077ed76d2)

**Comment by lnicola** [comment]
`/c:/dev/etc.` looks very wrong, but this is Windows-specific.

**Comment by Veykril** [comment]
Does r-a work just fine in the project if you dismiss and ignore the error? Then you don't need to provide anything, I noticed that bug as well (it randomly shows this on start up for me).

**Comment by lookbusy1344** [comment]
Yes, if I just ignore the pop up it works fine. It’s also fine if I click ‘no’. And it works fine if I click ‘yes’ and it adds the settings.json changes (although the pop up occurs again next time I open the project).

Thanks!

**Comment by lnicola** [comment]
I can reproduce this (or a similar issue) reliably by having `crates/hir-ty/src/tests.rs` open when I start Code. `./crates/hir-ty/Cargo.toml` gets added to `rust-analyzer.linkedProjects`. I'm running 8e581ac348, but I think it has been happening for a week or two.

**Comment by lookbusy1344** [comment]
Still a problem with 0.3.1916

**Comment by lookbusy1344** [comment]
This seems to be resolved now? Running v0.3.1950
